### ISSUE TITLE
Docker package page shows incorrect default suggestion

### ISSUE THREAD

**Issue by zanieb** [astral-sh/uv#7458]
<img width="783" alt="Screenshot 2024-09-17 at 8 20 00 AM" src="https://github.com/user-attachments/assets/19fe9fd0-e99c-44a3-a4d9-3e9760fc4d32">

I would expect at least the latest Python version? And probably not a specific variant?

This seems like it might be a pain to fix.

cc @samypr100

**Comment by samypr100** [comment]
The API shows them in the order they're published :(
We'd have to disable the matrix approach if we want some kind of order.

**Comment by zanieb** [comment]
Can we just publish the "right" one a second time after all the rest?

**Comment by samypr100** [comment]
Yes, I was thinking about that too, adding a final step to re-publish.

**Comment by samypr100** [comment]
I gave it a try and it didn't quite work, GitHub seems to sort based on when digests get pushed, so re-doing the `docker-publish` job at the end (which only updates the tags) didn't help and we'd have to rebuild a new image to have the digests change. It seems like a limitation on Github's side to not allow configurability of which tag gets featured. I'll keep thinking about it.

**Comment by samypr100** [comment]
> I gave it a try and it didn't quite work as I expected, GitHub seems to sort based on when digests get pushed, so re-doing the `docker-publish` job at the end (which only updates the tags) didn't help and we'd have to rebuild a new image to have the digests change. It seems like a limitation on Github's side to not allow configurability of which tag gets featured. I'll keep thinking about it.

I found a trick by leveraging annotations proposed in #7568 by which makes this achievable.
